From 00cb0406f38bc4f25e4ffbc85a3f2c12f6064878 Mon Sep 17 00:00:00 2001
From: Roger Quadros <rogerq@ti.com>
Date: Tue, 23 Jul 2013 12:31:05 +0300
Subject: [PATCH 14/14] temp: omap-usb-host: configre wakeup based on children

Signed-off-by: Roger Quadros <rogerq@ti.com>
---
 drivers/mfd/omap-usb-host.c |   23 ++++++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/omap-usb-host.c b/drivers/mfd/omap-usb-host.c
index 3734d16..13db62e 100644
--- a/drivers/mfd/omap-usb-host.c
+++ b/drivers/mfd/omap-usb-host.c
@@ -373,11 +373,23 @@ static int usbhs_runtime_resume(struct device *dev)
 	return 0;
 }
 
+static int usbhs_check_wakeup(struct device *dev, void *data)
+{
+	bool do_wakeup = device_may_wakeup(dev);
+	u8 *wakeup_data = data;
+
+	dev_info(dev, "wakeup %d\n", do_wakeup);
+	*wakeup_data |= do_wakeup;
+
+	return 0;
+}
+
 static int usbhs_runtime_suspend(struct device *dev)
 {
 	struct usbhs_hcd_omap		*omap = dev_get_drvdata(dev);
 	struct usbhs_omap_platform_data	*pdata = omap->pdata;
 	int i;
+	u8 wakeup;
 
 	dev_dbg(dev, "usbhs_runtime_suspend\n");
 
@@ -410,7 +422,15 @@ static int usbhs_runtime_suspend(struct device *dev)
 		clk_disable(omap->ehci_logic_fck);
 
 	omap_tll_disable(pdata);
-	pinctrl_pm_select_idle_state(dev);
+
+	/* Enable wakeup only if any of the children have wakeups enabled */
+	wakeup = 0;
+	device_for_each_child(dev, &wakeup, usbhs_check_wakeup);
+	if (wakeup)
+	{
+		dev_info(dev, "enabling wakeup\n");
+		pinctrl_pm_select_idle_state(dev);
+	}
 
 	return 0;
 }
@@ -815,6 +835,7 @@ static int usbhs_omap_probe(struct platform_device *pdev)
 	}
 
 	pinctrl_pm_select_default_state(dev);
+	device_init_wakeup(dev, true);
 
 	return 0;
 
-- 
1.7.10.4

